src/callbacks_step/glm_speed.jl

```diff
@@ -52,21 +53,18 @@ end
 # This method is called as callback after the StepsizeCallback during the time integration.
 @inline function (glm_speed_callback::GlmSpeedCallback)(integrator)
 
-  dt = integrator.dtcache
+  dt = get_proposed_dt(integrator)
   semi = integrator.p
-  mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
-  @unpack glm_scale, cfl_scale = glm_speed_callback
+  _, equations, solver, cache = mesh_equations_solver_cache(semi)
+  @unpack glm_scale, cfl = glm_speed_callback
 
   # compute time step for GLM linear advection equation with c_h=1 (redone due to the possible AMR)
-  max_scaled_speed_for_c_h = nextfloat(zero(dt))
-  for element in eachelement(solver, cache)
-    inv_jacobian = cache.elements.inverse_jacobian[element]
-    max_scaled_speed_for_c_h = max(max_scaled_speed_for_c_h, ndims(semi.equations) * inv_jacobian)
-  end
-  c_h_deltat = cfl_scale * 2 / (nnodes(solver) * max_scaled_speed_for_c_h)
+  c_h_deltat = calc_dt_for_cleaning_speed(cfl, equations, solver, cache)
```

Comment:
```suggestion
  c_h_deltat = calc_dt_for_cleaning_speed(cfl, mesh, equations, solver, cache)
```
I think we should keep passing the `mesh` here.
- It's consistent with the use in other funcitons.
- We can dispatch on the `mesh`, e.g. for MPi parallelism.

Reply:
okay, I will change this. I thought I was being clever avoiding the mesh but it does make sense for consistency for the callbacks to all use these extracted pieces from `semi`